test-data/unit/fine-grained.test

```diff
@@ -3448,6 +3448,30 @@ f(a.x)
 [out]
 ==
 
+[case testNamedTupleUpdate5]
```

Comment:
Can you also a coarse-grained incremental test? In particular, test that we can serialize recursive named tuples (both type infos and types in annotations).